### DISCUSSION TITLE
Dependency pipeline's outputs not available in dependent pipelines

### DISCUSSION THREAD

**Discussion by Perksey** [statiqdev/Discussions#95] [Q&A]
Hey there :wave:

I'm currently working on migrating the [Silk.NET](https://github.com/dotnet/Silk.NET) [WIP website](https://dotnet.github.io/Silk.NET) from [a custom-made tool](https://github.com/Perksey/Raisin) to Statiq Framework.

I've opted to use Statiq Framework instead of Statiq Web to keep full control over the generation process, which will prove useful later down the line when we add API documentation.

Currently we have 4 pipelines:
- **DirectCopying**: just copies CSS, JavaScript, images, and whatever else to the output directory.
- **FrontMatter**: Gets all of the actual content of the website and extracts its front matter
- **Markdown**: Sources that are recognised as markdown are rendered before going to...
- **Razor**: Renders content as a Razor page (using the layout)

Perhaps this is not the most optimal way of doing things, but in any case we're running into an issue where despite us declaring that **Markdown** is dependent on **FrontMatter**, we're getting no input documents for the former despite getting 16 outputs from the latter.

![image](https://user-images.githubusercontent.com/11160611/132986501-77446a6e-c445-4786-900f-8d9d9b78c9c9.png)

Any idea what could be going wrong?

Here's the generator code: https://github.com/dotnet/Silk.NET/blob/feature/heavy-wip-website-statiq-migration/src/Website/Silk.NET.Statiq/Program.cs

Thanks in advance for any help!

**Comment by daveaglick** [comment]
This is an easy one :). The issue is that while declaring a pipeline as having a dependency on another one makes the output documents of the former available to the latter, that's all it does. Basically if you don't make this declaration and try to access the document in any way, Statiq will throw. The missing piece is how to bring those documents into the dependent pipeline. Statiq doesn't make any assumptions about how you want to do this - you may want them available at the start of the pipeline (as it sounds like in this case), or you may just want to traverse them inside a Razor page or something without actually bringing them into the pipeline as documents.

If I'm understanding correctly and you want the dependent pipeline to start with the previous pipeline's outputs as it's own inputs, the easiest way is to add a `ReplaceDocuments` module as the first module in the dependent pipeline: `new ReplaceDocuments("FrontMatter")`. That'll stick take the outputs from the front matter pipeline and output them from the module, effectively making it the starting point for the follow-up pipeline.
